src/api-docs/source/index.html.md

```diff
@@ -45,18 +45,127 @@ search: true
+
+Network   | Wallet API Port | Node API Port
+--------- | --------------- | -------------
+main      | 12039           | 12037
+testnet   | 13039           | 13037
+regtest   | 14039           | 14037
+simnet    | 15039           | 15037
+
+In addition, each server has two API formats: JSON-RPC and a RESTful HTTP API.
+In general, the HTTP APIs respond to `GET` requests and the RPC APIs respond to
+`POST` requests. There are exceptions, so be sure to read the docs carefully.
+
+**Examples:**
+
+**Get a [transaction by hash](#get-tx-by-txhash) from the full node HTTP API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14037/tx/4674eb87021d9e07ff68cfaaaddfb010d799246b8f89941c58b8673386ce294f
+</code>
+
+**Get a [raw transaction by hash](#getrawtransaction) from the full node RPC API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14037
+  -X POST
+  --data '{
+    "method": "getrawtransaction",
+    "params": [ "4674eb87021d9e07ff68cfaaaddfb010d799246b8f89941c58b8673386ce294f"]
+  }'
+</code>
+
+**Get a [transaction by hash](#get-wallet-tx-details) from the "primary" wallet HTTP API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14039/wallet/primary/tx/4674eb87021d9e07ff68cfaaaddfb010d799246b8f89941c58b8673386ce294f
+</code>
+
+**Get a [transaction by hash](#gettransaction) from the wallet RPC API:**
+
+<code>
+curl http://x:api-key@127.0.0.1:14039
+  -X POST
+  --data '{
+    "method": "gettransaction",
+    "params": [ "4674eb87021d9e07ff68cfaaaddfb010d799246b8f89941c58b8673386ce294f" ]
+  }'
+</code>
+
+**Indexing**
+
+hsd has two indexer modules that are inactive by default. When turned on, the indexers
+enable additional API calls.
+
+Transaction indexer: enabled by `--index-tx=true`. Allows lookup of arbitrary transactions
+by their hash (txid). When disabled, the wallet still tracks all of its own transactions
+by hash and the full node can still be used to look up UTXO (unspent transaction outputs).
+
+Address indexer: enabled by `--index-address=true`. Allows lookup of all transactions
+involving a certain address.
+
+
+**Wallet: BIP44**
+
+The hsd wallet follows a [BIP44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki)
+structure with cointype `5353` for mainnet. A single hsd instance can have multiple BIP44
+wallets in its database.
+
+The hierarchy looks like this:
+
+`WalletDB`: A levelDB database stored in a `/wallet` directory, accessed by the wallet module
+plugged-in by default to the full node. Can process transactions for multiple wallets.
+
+`Wallet`: A BIP44 wallet based on 128 or 256 bits of entropy and backed up by a 12- or 24-
+word phrase. Multiple accounts can be derived from a single wallet.
+
+`Account`: A branch of a BIP44 wallet. Multiple addresses can be derived from a single
+account, and the wallet can be configured to spend coins from a single specified account.
+
+`Address`: A hash of a public key or script that can be given to a counterparty with the intent
+of receiving funds.
```

Comment:
There are two types of outputs defined for version 0 outputs. Pay-to-Witness-Pubkeyhash have a 20 byte hash of a public key and can be spent if the owner reveals the public key and provides a signature from the corresponding private key. Pay-to-Witness-Scripthash have a 32 byte hash of a Bitcoin script and can be spent if the owner reveals the script along with any additional data required to satisfy the script.